src/test/java/jenkins/plugins/git/GitToolChooserTest.java

```diff
@@ -65,6 +68,94 @@ public void enableSystemCredentialsProvider() {
         assertThat("The system credentials provider is enabled", store, notNullValue());
     }
 
+    /*
+    This test checks the GitToolChooser in a scenario where repo size>5M, user's choice is `jgit`.
+    In the event of having no node-specific installations, GitToolChooser will choose to return the default installation.
+     */
+    @Issue("JENKINS-63519")
+    @Test
+    public void testResolveGitTool() throws IOException, InterruptedException {
+        String remote = "https://gitlab.com/rishabhBudhouliya/git-plugin.git";
+        Item context = Mockito.mock(Item.class);
+        String credentialsId = null;
+
+        GitTool tool = new GitTool("my-git", SystemUtils.IS_OS_WINDOWS ? "git.exe" : "git", Collections.<ToolProperty<?>>emptyList());
+        GitTool JTool = new JGitTool(Collections.<ToolProperty<?>>emptyList());
+        jenkins.jenkins.getDescriptorByType(GitTool.DescriptorImpl.class).setInstallations(tool, JTool);
+
+        GitToolChooser r = new GitToolChooser(remote, context, credentialsId, JTool, null, TaskListener.NULL,true);
+
+        assertThat(r.getGitTool(), containsString(SystemUtils.IS_OS_WINDOWS ? "git.exe" : "git"));
+    }
+
+    /*
+    This test checks the GitToolChooser in a scenario where repo size>5M, user's choice is `jgit`.
+    There is no specific agent(node=null). In this case agent = Jenkins.get().
+    In the event of running the GitToolChooser on the agent, it should correctly predict the git installation for
+    that specific agent.
+     */
+    @Issue("JENKINS-63519")
+    @Test
+    public void testResolveGitToolWithJenkins() throws IOException, InterruptedException {
+        String remote = "https://gitlab.com/rishabhBudhouliya/git-plugin.git";
+        Item context = Mockito.mock(Item.class);
+        String credentialsId = null;
+
+        TestToolInstaller inst = new TestToolInstaller("master", "echo Hello", SystemUtils.IS_OS_WINDOWS ? "updated/git.exe" : "updated/git");
+        GitTool t = new GitTool("myGit", SystemUtils.IS_OS_WINDOWS ? "default/git.exe" : "default/git", Collections.singletonList(
```

Comment:
Also need to switch to use the `isWindows()` technique that is used elsewhere in the plugin.  Yes, it would be safe in the tests to use the Apache SystemUtils technique instead, but I'd rather have the same technique used throughout the plugin, both in the tests and in the production code.
```suggestion
        GitTool t = new GitTool("myGit", SystemUtils.IS_OS_WINDOWS ? "default\\git.exe" : "default/git", Collections.singletonList(
```